src/Router.ts

```diff
@@ -313,12 +313,17 @@ function applyTemplate(
     case 'function':
       return (template as TemplateFunction)(req, parentTemplate);
     case 'object':
-      return Object.keys(template).reduce((obj, key) => {
-        obj[key] = applyTemplate(
-          (template as TemplateObject)[key],
-          req,
-          template,
-        );
+      const _template = template as TemplateObject;
+      return Object.keys(_template).reduce((obj, key) => {
+        // undefined values cannot be set on headers or body
+        if ('undefined' !== typeof _template[key]) {
```

Comment:
seems overly complex

Will it work if you just return null if template is null at line 300?
```ts
  if (null === template) {
    return null;
  }

  if ('undefined' === typeof template) {
    return '';
  }
```

Reply:
Following our PM, null and undefined values are now returned as is.